app/controllers/course/assessment/submission/answer/answers_controller.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+class Course::Assessment::Submission::Answer::AnswersController < \
```

Comment:
Sorry I didn't catch this earlier, but I realised there is a `Course::Assessment::Submission::Answer::Programming::Controller` - would it be better to nest the controller method there?

Reply:
I was thinking this would be the best, instead of through a `...programming/files` controller, since the `create_programming_files` is done through the `Course::Assessment::Answer::Programming` model, and also updates other existing files at the same time. 

Just that because we use `Course::Assessment::Answer.id` on the front-end (not `Course::Assessment::Answer::Programming.id`). Additionally, since `programming` is just namespaced i thought it would be weird if the url was something like `.../submissions/:submission_id/answers/:id/programming/create_programming_file` (or is this alright?). So i left it at the `Course::Assessment::Submission::Answer::AnswersController`

Reply:
I actually think the url is all right. 

To be fair, controllers and urls can be made to deviate a little if it makes more sense. Grouping the controllers would be useful when you need to do similar functions, like check whether the answer if a programming one. For url, as long as it makes sense and do-able on front end that is ok. what you suggested is ok, if not `.../answers/:id/programming/file` (and use the usual the RESTFUL conventions).